### PR TITLE
 feat:support clustering algorithm

### PR DESCRIPTION
**Pull request by yizhihenpidehou**

### CODE REVIEW

**Comment by tddschn** [comment]
Hi @yizhihenpidehou, please add the `.idea` folder to your .gitignore file and use an descriptive PR title. Thanks. :)

File: myfile.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+y
```
Comment (tddschn) [inline myfile.txt:1]: What's this file?

Comment (yizhihenpidehou) [reply myfile.txt:1]: 不好意思，我测试git的时候新建的文件，忘记删除了！

**Comment by coreturn** [comment]
> Hi @yizhihenpidehou, please add the `.idea` folder to your .gitignore file and use an descriptive PR title. Thanks. :)

I have seen your commit message is descriptive, you can just name your PR title same as your commit message, which is easy to read. Refer to: https://www.ruanyifeng.com/blog/2016/01/commit_message_change_log.html

Btw, you can use the "git rebase" to merge your two commits into one. Refer to: https://cloud.tencent.com/developer/article/1690638

Overall, work is pretty good, solve these comments and push the code again~

**Pull request merged**